### PR TITLE
Implemetation of the API call

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
The API call was implemented

Postman
image
{
            "dt": 1628629200,
            "main": {
                "temp": 20.44,
                "feels_like": 20.99,
                "temp_min": 19.17,
                "temp_max": 20.44,
                "pressure": 1006,
                "sea_level": 1006,
                "grnd_level": 833,
                "humidity": 94,
                "temp_kf": 1.27
            },
            "weather": [
                {
                    "id": 500,
                    "main": "Rain",
                    "description": "light rain",
                    "icon": "10n"
                }
            ],
            "clouds": {
                "all": 90
            },
            "wind": {
                "speed": 0.34,
                "deg": 45,
                "gust": 0.53
            },
            "visibility": 10000,
            "pop": 0.77,
            "rain": {
                "3h": 1.75
            },
            "sys": {
                "pod": "n"
            },
            "dt_txt": "2021-08-10 21:00:00"
        },
        {
            "dt": 1628640000,
            "main": {
                "temp": 19.84,
                "feels_like": 20.31,
                "temp_min": 19.22,
                "temp_max": 19.84,
                "pressure": 1005,
                "sea_level": 1005,
                "grnd_level": 834,
                "humidity": 93,
                "temp_kf": 0.62
            },
            "weather": [
                {
                    "id": 500,
                    "main": "Rain",
                    "description": "light rain",
                    "icon": "10d"
                }
            ],
            "clouds": {
                "all": 93
            },
            "wind": {
                "speed": 0.4,
                "deg": 28,
                "gust": 0.49
            },
            "visibility": 10000,
            "pop": 0.74,
            "rain": {
                "3h": 1.01
            },
            "sys": {
                "pod": "d"
            },
            "dt_txt": "2021-08-11 00:00:00"
        },
        {
            "dt": 1628650800,
            "main": {
                "temp": 20.55,
                "feels_like": 21.01,
                "temp_min": 20.55,
                "temp_max": 20.55,
                "pressure": 1007,
                "sea_level": 1007,
                "grnd_level": 836,
                "humidity": 90,
                "temp_kf": 0
            },
            "weather": [
                {
                    "id": 500,
                    "main": "Rain",
                    "description": "light rain",
                    "icon": "10d"
                }
            ],
            "clouds": {
                "all": 100
            },
            "wind": {
                "speed": 0.67,
                "deg": 222,
                "gust": 0.77
            },
            "visibility": 10000,
            "pop": 0.72,
            "rain": {
                "3h": 0.86
            },
            "sys": {
                "pod": "d"
            },
            "dt_txt": "2021-08-11 03:00:00"
        }

Android Studio
image

image

image

### CODE REVIEW

File: app/src/main/java/com/example/rxandroid/data/api/WeatherAPI.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.example.rxandroid.data.api
+
+import com.example.rxandroid.data.response.JSONResponse
+import retrofit2.Call
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface WeatherAPI {
+
+    @GET("onecall?")
```
Comment (BQuinteros) [inline app/src/main/java/com/example/rxandroid/data/api/WeatherAPI.kt]: I think this is not the correct API we call

Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/rxandroid/data/api/WeatherAPI.kt]: done

File: app/src/main/java/com/example/rxandroid/data/mapper/WeatherMapper.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import com.example.rxandroid.domain.entity.DayEntity
+import com.example.rxandroid.domain.entity.ForecastApiEntity
+import com.example.rxandroid.domain.entity.MainEntity
+import com.example.rxandroid.domain.entity.RainEntity
+import com.example.rxandroid.domain.entity.SysEntity
+import com.example.rxandroid.domain.entity.WeatherEntity
+import com.example.rxandroid.domain.entity.WindEntity
+
+
+fun ForecastApiResponse.transform() = ForecastApiEntity(
```
Comment (SartiMau) [inline app/src/main/java/com/example/rxandroid/data/mapper/WeatherMapper.kt:24]: all these functions should be public?

File: app/src/main/java/com/example/rxandroid/data/mapper/WeatherMapper.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import com.example.rxandroid.domain.entity.CoordEntity
+import com.example.rxandroid.domain.entity.DayEntity
+import com.example.rxandroid.domain.entity.ForecastApiEntity
+import com.example.rxandroid.domain.entity.MainEntity
+import com.example.rxandroid.domain.entity.RainEntity
+import com.example.rxandroid.domain.entity.SysEntity
+import com.example.rxandroid.domain.entity.WeatherEntity
+import com.example.rxandroid.domain.entity.WindEntity
+
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/rxandroid/data/mapper/WeatherMapper.kt]: delete this extra line pls

File: app/src/main/java/com/example/rxandroid/data/service/WeatherService.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        if (response.isSuccessful) {
+            subscriber.onNext(response.body()?.transform())
+        } else {
+            subscriber.onError(Throwable(response.message()))
+        }
+    }
+
+    companion object {
+        private const val CITY_NAME: String = "Tandil"
+        private const val APP_KEY: String = "46a38f32b732c581ed6e50a148147864"
```
Comment (SartiMau) [inline app/src/main/java/com/example/rxandroid/data/service/WeatherService.kt]: put this key in a gradle variable pls

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/rxandroid/data/mapper/WeatherMapper.kt (hunk 3 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/rxandroid/data/mapper/WeatherMapper.kt]: done

File: app/src/main/java/com/example/rxandroid/data/mapper/WeatherMapper.kt (hunk 2 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/rxandroid/data/mapper/WeatherMapper.kt:24]: changed

File: app/src/main/java/com/example/rxandroid/data/service/WeatherService.kt (hunk 4 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/rxandroid/data/service/WeatherService.kt]: done

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Pull request merged**